### PR TITLE
dm: fix lag keeps growing when failed ddl is skipped and no following ddl

### LINKED ISSUE #9605: replication lag not refreshed when last ddl is failed and skipped, and no futher ddl
### What did you do?

- start all mode task
- in sync phase, run a ddl in upstream, and this ddl will fail on downstream
- task will paused, skip the ddl, then start task

### What did you expect to see?

lag will down to 0

### What did you see instead?

lag keep increasing

### Versions of the cluster

DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):
```console
v5.4.1
```
Upstream MySQL/MariaDB server version:
```console
(paste upstream MySQL/MariaDB server version here)
```
Downstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):
```console
(paste TiDB cluster version here)
```
How did you deploy DM: tiup or manually?
```console
(leave TiUP or manually here)
```
Other interesting information (system version, hardware config, etc):
```console
>
>
```

### current status of DM cluster (execute `query-status <task-name>` in dmctl)

```console
(paste current status of DM cluster here)
```

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: close #9605

### What is changed and how it works?
- reset recorded ts on syncer restart

#### Tests 

 - [x] Unit test
 - Integration test
 - [x] Manual test (add detailed scripts or steps below)
  - test using  steps in the issue, before the change, lag keeps growing, became 0 after this pr
```
before 

➜  dm curl -s "127.0.0.1:8361/metrics" | grep -v '#'| grep dm_syncer_replication_lag_gauge
dm_syncer_replication_lag_gauge{source_id="mysql-3307",task="test",worker="worker-1"} 303
➜  dm curl -s "127.0.0.1:8361/metrics" | grep -v '#'| grep dm_syncer_replication_lag_gauge
dm_syncer_replication_lag_gauge{source_id="mysql-3307",task="test",worker="worker-1"} 313


after
➜  dm curl -s "127.0.0.1:8361/metrics" | grep -v '#'| grep dm_syncer_replication_lag_gauge
dm_syncer_replication_lag_gauge{source_id="mysql-3307",task="test",worker="worker-1"} 0
➜  dm curl -s "127.0.0.1:8361/metrics" | grep -v '#'| grep dm_syncer_replication_lag_gauge
dm_syncer_replication_lag_gauge{source_id="mysql-3307",task="test",worker="worker-1"} 0
```

 - No code

### Release note 

```release-note
fix lag keeps growing when failed ddl is skipped and no following ddl
```

### CODE REVIEW

File: dm/syncer/syncer.go [hunk 1]
```diff
@@ -913,11 +929,6 @@ func (s *Syncer) updateReplicationLagMetric() {
 			s.tctx.L().Info("ShowLagInLog", zap.Int64("lag", lag))
 		}
 	})
-
-	// reset skip job TS in case of skip job TS is never updated
-	if minTS == s.workerJobTSArray[skipJobIdx].Load() {
```
Comment (GMHDBJD) [inline dm/syncer/syncer.go:918]: why remove these lines

Comment (D3Hunter) [reply dm/syncer/syncer.go:918]: skipped event shouldn't be accounted when calculating lag

and it's not used elsewhere

Comment (D3Hunter) [reply dm/syncer/syncer.go:918]: but this do change `lag metric` behavior when syncer is catching up and the distance between up and downstream is quite long, and there's no interested binlog event in between.
before this pr, there's be a decreasing lag, but 0 after.

will revert it, should change it in another pr

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Review by lance6716 — APPROVED** [review]

**Pull request merged**